core/controllers/profile_test.py

```diff
@@ -282,3 +284,32 @@ def test_profile_data_is_independent_of_currently_logged_in_user(self):
         response = self.get_json(
             '/profilehandler/data/%s' % self.EDITOR_USERNAME)
         self.assertEqual(response['user_bio'], 'My new editor bio')
+
+
+class FirstContributionDateTests(test_utils.GenericTestBase):
+
+    USERNAME = 'abc123'
+    EMAIL = 'abc123@gmail.com'
+
+    def test_contribution_datetime(self):
+        """Test the contribution date shows up correctly as nonexist."""
+        self.signup(self.EMAIL, self.USERNAME)
+        self.login(self.EMAIL)
+        user_services.get_or_create_user(self.USERNAME, self.EMAIL)
+        self.user_id = self.get_user_id_from_email(self.EMAIL)
+        response = self.testapp.get('/profile/%s' % self.USERNAME)
+        self.assertEqual(response.status_int, 200)
+        response_dict = self.get_json(
+            '/profilehandler/data/%s' % self.USERNAME)
+        self.assertEqual(response_dict['first_contribution_datetime'], None)
+
+        current_datetime = datetime.datetime.utcnow()
+        user_services.update_first_contribution_datetime(self.user_id, current_datetime)
```

Comment:
Here and below: lines should be at most 80 chars.

If you are using Sublime, you can add this to your Preferences > Settings - User; among other things, it includes a ruler at the 80-character mark:

{
	"color_scheme": "Packages/User/SublimeLinter/Monokai (SL).tmTheme",
	"ensure_newline_at_eof_on_save": true,
	"font_size": 9,
	"highlight_line": true,
	"rulers":
	[
		80
	],
	"shift_tab_unindent": true,
	"spell_check": true,
	"tab_size": 4,
	"translate_tabs_to_spaces": true,
	"trim_trailing_white_space_on_save": true,
	"update_check": false
}